tests/src/core/importType.js

```diff
@@ -42,20 +43,35 @@ describe('importType(name)', function () {
   })
 
   it("should return 'internal' for non-builtins resolved outside of node_modules", function () {
-    const pathContext = testContext({ "import/resolver": { node: { paths: [ path.join(__dirname, '..', '..', 'files') ] } } })
+    const pathContext = testContext({ "import/resolver": { node: { paths: [pathToTestFiles] } } })
     expect(importType('importType', pathContext)).to.equal('internal')
   })
 
   it.skip("should return 'internal' for scoped packages resolved outside of node_modules", function () {
-    const pathContext = testContext({ "import/resolver": { node: { paths: [ path.join(__dirname, '..', '..', 'files') ] } } })
+    const pathContext = testContext({ "import/resolver": { node: { paths: [pathToTestFiles] } } })
     expect(importType('@importType/index', pathContext)).to.equal('internal')
   })
     
   it("should return 'internal' for internal modules that are referenced by aliases", function () {
-    const pathContext = testContext({ 'import/resolver': { node: { paths: [path.join(__dirname, '..', '..', 'files')] } } })
+    const pathContext = testContext({ 'import/resolver': { node: { paths: [pathToTestFiles] } } })
     expect(importType('@my-alias/fn', pathContext)).to.equal('internal')
   })
 
+  it("should return 'internal' for aliased internal modules that look like core modules (node resolver)", function () {
+    const pathContext = testContext({ 'import/resolver': { node: { paths: [pathToTestFiles] } } })
+    expect(importType('constants/index', pathContext)).to.equal('internal')
+    // the following assertion fails because the node resolver runs an resolve.isCore('constants') without first checking paths config
+    // expect(importType('constants', pathContext)).to.equal('internal')
```

Comment:
With my changes, this will still return `'builtin'` unless changes are made to the node resolver.

Reply:
In particular: https://github.com/benmosher/eslint-plugin-import/blob/master/resolvers/node/index.js#L12-L15

Reply:
Can that be changed in this PR? We can always merge, release the node resolver first, and release the main plugin after.

Reply:
I did a little local testing and I think the current behavior here is actually correct. If you `require('constants')` in node, it appears to always resolve to the core module rather than anything matching NODE_PATH. `require('constants/index')` however, does resolve the local file. For that reason, I think it's better to leave the node resolver as-is.

Reply:
Similarly, `require(‘constants/‘)` should always match the local file - if that’s the case then this is fine.

Reply:
That does appear to be the case. Added a test.